lib/_http_client.js

```diff
@@ -356,7 +356,9 @@ function socketCloseListener() {
     // Socket closed before we emitted 'end' below.
     if (!res.complete) {
       res.aborted = true;
-      res.emit('aborted');
+      if (!res.emit('aborted')) {
+        res.emit('error', connResetException('socket hang up'));
```

Comment:
This error was unique, in the sense that it was only emitted when we did not have a `response`, can something different be used for this?

Reply:
I disagree. I don't think I quite understand your argument? How is ECONNRESET unique? I think it makes total sense here. Instead of the request being "aborted" it's moved to the response.

https://www-numi.fnal.gov/offline_software/srt_public_context/WebDocs/Errors/unix_system_errors.html

> #define ECONNRESET      104     /* Connection reset by peer */

Reply:
I agree with @lpinca. I prefer we added a new error here.

Reply:
I'm not talking about `ECONNRESET` but "socket hang up".

Reply:
Any suggestions?

Reply:
`ECONNABORTED`?